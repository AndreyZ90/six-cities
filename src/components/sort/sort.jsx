import React from 'react';
import PropTypes from 'prop-types';

import SortItem from '@/components/sort-item/sort-item';

import withActiveFlag from '@/HOCs/with-active-flag/with-active-flag';

import {SortType} from '@/helpers/const';

const Sort = (props) => {
  const {currentSort, isActiveFlag, currentCity, onFlagChange} = props;

  const activeClass = isActiveFlag ? `places__options--opened` : ``;

  return (
    <form className="places__sorting" action="#" method="get">
      <span className="places__sorting-caption">Sort by</span>
      <span className="places__sorting-type" tabIndex={0} onClick={onFlagChange}>
        {currentSort}
        <svg className="places__sorting-arrow" width={7} height={4}>
          <use xlinkHref="#icon-arrow-select" />
        </svg>
      </span>
      <ul className={`places__options places__options--custom ${activeClass}`}>
        {Object.values(SortType).map((item) => (
          <SortItem
            key={item}
            item={item}
            isActive={currentSort === item}
            currentCity={currentCity}
            onFlagChange={onFlagChange}
          />
        ))}
      </ul>
    </form>
  );
};

Sort.propTypes = {
  currentSort: PropTypes.oneOf(Object.values(SortType)).isRequired,
  isActiveFlag: PropTypes.bool.isRequired,
  currentCity: PropTypes.string.isRequired,
  onFlagChange: PropTypes.func.isRequired
};

export {Sort};
export default withActiveFlag(Sort);

