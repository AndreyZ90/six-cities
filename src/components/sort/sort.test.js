import React from 'react';
import TestRenderer from 'react-test-renderer';
import {BrowserRouter as Router} from 'react-router-dom';

import {Sort} from '@/components/sort/sort';

const currentSort = `Popular`;
const currentCity = `Amsterdam`;

describe(`Sort component snapshot`, () => {
  test(`Should correctly render Sort component`, () => {
    const tree = TestRenderer.create(
        <Router>
          <Sort
            currentSort={currentSort}
            isActiveFlag={false}
            currentCity={currentCity}
            onFlagChange={() => {}}
          />
        </Router>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
