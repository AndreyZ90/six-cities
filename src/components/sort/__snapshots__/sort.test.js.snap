// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort component snapshot Should correctly render Sort component 1`] = `
<form
  action="#"
  className="places__sorting"
  method="get"
>
  <span
    className="places__sorting-caption"
  >
    Sort by
  </span>
  <span
    className="places__sorting-type"
    onClick={[Function]}
    tabIndex={0}
  >
    Popular
    <svg
      className="places__sorting-arrow"
      height={4}
      width={7}
    >
      <use
        xlinkHref="#icon-arrow-select"
      />
    </svg>
  </span>
  <ul
    className="places__options places__options--custom "
  >
    <a
      href="/amsterdam?sort=Popular"
      onClick={[Function]}
    >
      <li
        className="places__option places__option--active"
        onClick={[Function]}
        tabIndex={0}
      >
        Popular
      </li>
    </a>
    <a
      href="/amsterdam?sort=Price: low to high"
      onClick={[Function]}
    >
      <li
        className="places__option "
        onClick={[Function]}
        tabIndex={0}
      >
        Price: low to high
      </li>
    </a>
    <a
      href="/amsterdam?sort=Price: high to low"
      onClick={[Function]}
    >
      <li
        className="places__option "
        onClick={[Function]}
        tabIndex={0}
      >
        Price: high to low
      </li>
    </a>
    <a
      href="/amsterdam?sort=Top rated first"
      onClick={[Function]}
    >
      <li
        className="places__option "
        onClick={[Function]}
        tabIndex={0}
      >
        Top rated first
      </li>
    </a>
  </ul>
</form>
`;
